src/common/download.h

```diff
@@ -32,5 +32,11 @@
 
 namespace tools
 {
-  bool download(const std::string &path, const std::string &url);
+  typedef void *download_async_handle;
```

Comment:
Also, what would the cleanup function do ? I don't actually have any custom code on deletion right now, and I don't see what would need adding with this new method.